[IGA] Adding topology items and python scripts

tteschemacher opened the pull request:
I am adding here the data structure for the brep geometry topology object for Nurbs-based CAD simulations. 

Additionally, the necessary Python scripts are added to allow the standard way to simulate with IGA similiar to the Structural Mechanics Application

tteschemacher committed:
New Topology objects for IGA-Brep Application

tteschemacher committed:
Some updates and new Face Class

tteschemacher committed:
Python scripts for IGA Analysis

tteschemacher committed:
full brep description

tteschemacher committed:
Updates in brep_topology objects

tteschemacher committed:
Updates in Brep Topology objects

philbucher commented on applications/IgaApplication/python_scripts/iga_analysis.py:
```diff
@@ -0,0 +1,73 @@
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, model, project_parameters):
```
you can remove this constructor. I don't think you will use MPI anytime soon ...

philbucher commented on applications/IgaApplication/python_scripts/iga_solver.py:
```diff
@@ -0,0 +1,464 @@
+    def ImportModelPart(self):
+        """This function imports the ModelPart
+        """
+        self._ImportModelPart(self.main_model_part, self.settings["model_import_settings"])
```
I am quite sure this does not work for you

philbucher commented on applications/IgaApplication/python_scripts/iga_solver.py:
```diff
@@ -0,0 +1,464 @@
+        name_out_file = self.settings["model_import_settings"]["input_filename"].GetString()+".out"
+        file = open(name_out_file + ".mdpa","w")
+        file.close()
+        KratosMultiphysics.ModelPartIO(name_out_file, KratosMultiphysics.IO.WRITE).WriteModelPart(self.main_model_part)
```
does this work in your case?

philbucher commented on applications/IgaApplication/python_scripts/iga_solver.py:
```diff
@@ -0,0 +1,464 @@
+            # Add constitutive laws and material properties from json file to model parts.
+            material_settings = KratosMultiphysics.Parameters("""{"Parameters": {"materials_filename": ""}} """)
+            material_settings["Parameters"]["materials_filename"].SetString(materials_filename)
+            KratosMultiphysics.ReadMaterialsUtility(material_settings, self.model)
```
Are you using this?

tteschemacher replied:
I do. Materials are the same as in the Structural Mechanics

philbucher commented on applications/IgaApplication/python_scripts/iga_static_solver.py:
```diff
@@ -0,0 +1,60 @@
+        self.print_on_rank_zero("::[StaticIgaSolver]:: ", "Finished initialization.")
+
+    def Solve(self):
+        super(StaticIgaSolver, self).Solve()
```
this fct does nothing and can be removed

philbucher commented on applications/IgaApplication/python_scripts/iga_static_solver.py:
```diff
@@ -0,0 +1,60 @@
+    def _create_solution_scheme(self):
+        return KratosMultiphysics.ResidualBasedIncrementalUpdateStaticScheme()
+
+    def _create_iga_solution_strategy(self):
```
why is this needed here => already exists in baseclass

philbucher commented on applications/IgaApplication/python_scripts/python_solvers_wrapper_iga.py:
```diff
@@ -0,0 +1,80 @@
+import KratosMultiphysics
+
+
+def CreateSolver(model, custom_settings):
```
this file is a copy from StructuralMechanics, you don't have most of the things ...

Also it is not up-to-date with the latest developments

tteschemacher replied:
Yes, you are right. That was the wrong version

philbucher reviewed (changes requested):
I suggest to split this PR into two parts, one for the python-scripts and one for the C-stuff

there is a lot of stuff just copied from StructuralMechanics, we should discuss it in person

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp:
```diff
@@ -0,0 +1,68 @@
+//
+//  Main authors:    Tobias Teschemacher
+//                   Michael Breitenberger
+//                   Thomas Oberbichler
```
You can drop my name since I haven't done nothing inside this file and I already have a lot of other stuff to maintain :P

tteschemacher replied:
Haha you can not escape from this.

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp:
```diff
@@ -0,0 +1,68 @@
+        int rBrepId,
+        std::vector<EdgeTopology>& rBrepEdgeTopologyVector,
+        std::vector<TrimmingRange>& rTrimmingRangeVector,
+        int& rDegree,
```
Maybe recheck `type&` vs `const type&` vs `const type`. Just for code quality.

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.h:
```diff
@@ -0,0 +1,141 @@
+//
+//  Main authors:    Tobias Teschemacher
+//                   Michael Breitenberger
+//                   Thomas Oberbichler
```
idem

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_model.cpp:
```diff
@@ -0,0 +1,38 @@
+//
+//  Main authors:    Tobias Teschemacher
+//                   Michael Breitenberger
+//                   Thomas Oberbichler
```
idem

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_trimming_curve.h:
```diff
@@ -0,0 +1,73 @@
+
+namespace Kratos
+{
+    using namespace ANurbs;
```
Maybe dangerous. Everything is imported into the Kratos namespace. You can put the `using` inside the function bodies.

tteschemacher replied:
I see the problem. Is there any other option to include Curve2D and CurveGeometry2D?

oberbichler replied:
Using `ANurbs::Curve2D` or put the `using` inside your function body. Anyway I have to add an alias for these entities into Kratos. After that we only need to change `ANurbs::Curve...` to `Curve...`

oberbichler commented on applications/IgaApplication/custom_utilities/nurbs_brep_modeler.cpp:
```diff
@@ -0,0 +1,36 @@
+//           Kratos default license: kratos/IGAStructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Tobias Tescheamacher
+//                   Thomas Oberbichler
```
idem

oberbichler commented on applications/IgaApplication/iga_application_variables.cpp:
```diff
@@ -13,9 +13,12 @@
 namespace Kratos
 {
 
+KRATOS_CREATE_VARIABLE(double, FACE_BREP_ID)
```
`double` or `int`?

tteschemacher replied:
changed it

oberbichler commented on applications/IgaApplication/custom_utilities/brep_topology/brep_face.cpp:
```diff
@@ -0,0 +1,72 @@
+        int number_of_nodes_u = rKnotVectorU.size() + rP - 1;
+        int number_of_nodes_v = rKnotVectorV.size() + rQ - 1;
+
+        m_node_surface_geometry_3d = New<NodeSurfaceGeometry3D>(
```
Instead of `ANurbs::New` you can use `Kratos::make_unique`. It does exactly the same. You can check out [shared_pointers.h](https://github.com/KratosMultiphysics/Kratos/blob/48be4aef49bab6dc4288d71f64b2044a1c1f3748/kratos/includes/shared_pointers.h#L41) for some helper functions in regard to smart pointers.

tteschemacher committed:
update of python_solvers_wrapper_iga.py

tteschemacher committed:
Deletion of __init__ - using directly the base function

tteschemacher committed:
Change double to int in FACE_BREP_ID

tteschemacher committed:
change New->Kratos::make_unique

tteschemacher committed:
Codacy updates

tteschemacher committed:
Read in of geometries for JSON geometries.

tteschemacher committed:
Updates to read in and create integration domain for brep edges.

philbucher committed:
solvers_wrapper update

philbucher committed:
updated static solver

philbucher committed:
updated iga-solver

tteschemacher committed:
bug fixes in iga_solver.py

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

tteschemacher committed:
Bug fixes to run IGA Analysis.

tteschemacher committed:
assign_vector_variable_and_constraints_to_conditions_process.py condition based on assign_vector_variable_to_conditions_process.py to set flags for constraints on conditions.

tteschemacher committed:
Updates to allow ImportModelPart for IGA analysis

tteschemacher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp:
```diff
@@ -0,0 +1,151 @@
+                    condition->SetValue(SHAPE_FUNCTION_VALUES, N_0);
+                    condition->SetValue(SHAPE_FUNCTION_LOCAL_DERIVATIVES, N_1);
+                    condition->SetValue(INTEGRATION_WEIGHT, integration_points[j].weight);
+                }
```
@philbucher, do you see the possibility to somehow template the condition and element?

tteschemacher committed:
Updates in Application

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.h:
```diff
@@ -0,0 +1,71 @@
+#if !defined(KRATOS_BREP_JSON_IO_H_INCLUDED )
```
please use the standard-header

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.h:
```diff
@@ -0,0 +1,71 @@
+
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "includes/node.h"
```
not needed

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+
+// Project includes
+#include "brep_json_io.h"
+#include "iga_application.h"
```
please remove this

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+        file.open(rOutputFileName);
+        std::string separator = "";
+        file << "{\"geometry_integration_points\" :[";
+        for (auto element = rModelPart.ElementsBegin(); element != rModelPart.ElementsEnd(); ++element)
```
`for (const auto& r_elem : rModelPart.Elements())`

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+        ModelPart& model_part,
+        Parameters& rNurbsBrepGeometryJson)
+    {
+        std::cout << "\n> Start reading CAD geometry" << std::endl;
```
Use the Kratos-Logger for this

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+
+    std::vector<BrepModel> BrepJsonIO::ImportNurbsBrepGeometry(
+        ModelPart& model_part,
+        Parameters& rNurbsBrepGeometryJson)
```
Parameters should be passed by-value

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+    {
+        std::cout << "\n> Start reading CAD geometry" << std::endl;
+
+        double model_tolerance = rNurbsBrepGeometryJson["tolerances"]["model_tolerance"].GetDouble();
```
const

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+
+        double model_tolerance = rNurbsBrepGeometryJson["tolerances"]["model_tolerance"].GetDouble();
+
+        std::vector<BrepModel> r_brep_model_vector;
```
`brep_model_vector`, this is not a reference

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+
+        std::vector<BrepModel> r_brep_model_vector;
+
+        for (int brep_i = 0; brep_i < rNurbsBrepGeometryJson["breps"].size(); brep_i++)
```
`std::size_t`

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+
+        for (int brep_i = 0; brep_i < rNurbsBrepGeometryJson["breps"].size(); brep_i++)
+        {
+            Parameters brep_json = rNurbsBrepGeometryJson["breps"][brep_i];
```
const auto

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+        for (int brep_i = 0; brep_i < rNurbsBrepGeometryJson["breps"].size(); brep_i++)
+        {
+            Parameters brep_json = rNurbsBrepGeometryJson["breps"][brep_i];
+            int brep_brep_id = brep_json["brep_id"].GetInt();
```
cpnst

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.cpp:
```diff
@@ -0,0 +1,454 @@
+        file.close();
+    }
+
+    std::vector<BrepModel> BrepJsonIO::ImportNurbsBrepGeometry(
```
![2mmfnh](https://user-images.githubusercontent.com/25484702/48555314-2290ca00-e8e1-11e8-802d-bcae7c930569.jpg)

Please refactor this into smaller fcts, this is ~400 lines long!

E.g. into edges, faces etc
Please take into account the comment I made abt other parts of the code

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_boundary_loop.cpp:
```diff
@@ -0,0 +1,43 @@
+
+    bool& BrepBoundaryLoop::IsOuterLoop()
+    {
+        return m_is_outer_loop;
```
members should be `mIsOuterLoop`

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_boundary_loop.cpp:
```diff
@@ -0,0 +1,43 @@
+        return m_brep_trimming_curves;
+    }
+
+    bool& BrepBoundaryLoop::IsOuterLoop()
```
why by-ref?

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_boundary_loop.h:
```diff
@@ -0,0 +1,61 @@
+#if !defined(KRATOS_BREP_BOUNDARY_LOOP_H_INCLUDED )
```
header

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_boundary_loop.h:
```diff
@@ -0,0 +1,61 @@
+// Project includes
+#include "brep_trimming_curve.h"
+
+#include "iga_application.h"
```
pls remove this

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_boundary_loop.h:
```diff
@@ -0,0 +1,61 @@
+
+        std::vector<BrepTrimmingCurve> GetTrimmingCurves();
+        bool& IsOuterLoop();
+        //important function, to be implemented!!
```
???

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp:
```diff
@@ -0,0 +1,151 @@
+    */
+    bool BrepEdge::IsCouplingEdge()
+    {
+        if (m_brep_edge_topology_vector.size() > 1)
```
this can be simplified to:
`return m_brep_edge_topology_vector.size() > 1`

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.cpp:
```diff
@@ -0,0 +1,151 @@
+                {
+                    int id = 0;
+                    if (rModelPart.GetRootModelPart().Elements().size()>0)
+                        id = rModelPart.GetRootModelPart().Elements().back().Id() + 1;
```
da rollt es mir ja die Fussnägel auf!
`id = rModelPart.GetRootModelPart().NumberOfElements() + 1`

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.h:
```diff
@@ -0,0 +1,160 @@
+#include "iga_application.h"
+#include "iga_application_variables.h"
+
+#include "../node_curve_geometry_3d.h"
```
no relative includes pls

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_edge.h:
```diff
@@ -0,0 +1,160 @@
+// System includes
+
+// Project includes
+#include "iga_application.h"
```
remove pls

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_face.cpp:
```diff
@@ -0,0 +1,251 @@
+//                   Multi-Physics
+//
+//  License:     BSD License
+//           Kratos default license: kratos/IGAStructuralMechanicsApplication/license.txt
```
this is no longer IGAStructuralMechanics => please point to the one in core

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_face.cpp:
```diff
@@ -0,0 +1,251 @@
+                        N_2(n, 2) = shape(4, indexU, indexV);
+                    }
+
+                    KRATOS_WATCH(N_0)
```
leftover debugging output

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_face.cpp:
```diff
@@ -0,0 +1,251 @@
+    {
+        //auto clipper = ANurbs::Tri();
+
+        Properties::Pointer this_property = rModelPart.pGetProperties(rPropertiesId);
```
why do you need a pointer?

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_model.cpp:
```diff
@@ -0,0 +1,87 @@
+// Project includes
+#include "brep_model.h"
+
+#include "iga_application.h"
```
remove

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_trimming_curve.cpp:
```diff
@@ -0,0 +1,66 @@
+
+namespace Kratos
+{
+    int& BrepTrimmingCurve::GetTrimIndex()
```
why reference?

philbucher commented on applications/IgaApplication/custom_utilities/brep_topology/brep_vertex.cpp:
```diff
@@ -0,0 +1,21 @@
+//    |  /           |
```
please remove empty files

unless you will fill it later

philbucher commented on applications/IgaApplication/custom_utilities/brep_json_io.h:
```diff
@@ -0,0 +1,71 @@
+    ///@}
+    ///@name Life Cycle
+    ///@{
+    void WriteIntegrationDomainJson(
```
I think this could be free functions, no need to put them in a class

philbucher reviewed (commented):
I started going through the C-code, but it is too much to look at at once

I gave you a bunch of suggestion for cleaning, please apply them in all files even though I did not point out all locations.

BTW I strongly suggest to create C-tests for the BREP-classes, especially considering the upcoming AMatrix-transition

Ofc it would also be good to have tests for other features like your Brep-json reader
There you should think abt cases in which it can fail => throw a meaningful message in those cases => check in the tests if these messages are thrown
I can show you how o do this

philbucher commented:
@tteschemacher also take a look at the codacy-suggestions, they are necessary

tteschemacher committed:
changes in topology objects

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

oberbichler commented:
@tteschemacher Please use cherrypick and don't copy&paste the code from other commits. Otherwise we will get in trouble when merging this thing.

philbucher commented:
> @tteschemacher Please use cherrypick and don't copy&paste the code from other commits. Otherwise we will get in trouble when merging this thing.

:D

![704e645b-d074-4c66-912f-ba48cff8b769](https://user-images.githubusercontent.com/25484702/48589549-446b6a80-e93c-11e8-86aa-4854aecefcfc.jpeg)

tteschemacher committed:
Merge remote-tracking branch 'origin/iga/anurbs-alias' into feature/iga_brep_topology_kernel

philbucher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

philbucher committed:
Merge branch 'iga/some-cleanup' into feature/iga_brep_topology_kernel

philbucher committed:
removing unused includes

philbucher committed:
wrong cmake include

philbucher committed:
[skip ci] minor

philbucher commented:
@tteschemacher I pushed some cleanup
this includes #3414 which should be merged before

philbucher committed:
using directly the logger

philbucher committed:
using proper way for getting the flag

oberbichler committed:
Add alias for PointOnCurveProjection and TrimmedSurfaceClipping

tteschemacher committed:
IGA Application updates

tteschemacher committed:
Merge branch 'feature/iga_brep_topology_kernel' of https://github.com/KratosMultiphysics/Kratos into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/IgaApplication/CMakeLists.txt
#	applications/IgaApplication/custom_utilities/brep_topology/brep_edge.h
#	applications/IgaApplication/custom_utilities/brep_topology/brep_face.h
#	applications/IgaApplication/custom_utilities/brep_topology/brep_model.h

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

tteschemacher committed:
Merge branch 'iga/anurbs-alias' into feature/iga_brep_topology_kernel

tteschemacher committed:
small fixes

tteschemacher committed:
external_libraries -  push to be deleted

tteschemacher committed:
Updates to allow integration of CAD geometry.

tteschemacher committed:
Iga output process to write output on integration points respectively discrete integration points/ elements.

tteschemacher committed:
Bugfix: self.sub_model_part -> self.model_part

tteschemacher committed:
Update towards GiD output process style

Conflicts:
	applications/IgaApplication/python_scripts/iga_io_process.py

philbucher committed:
some updates

philbucher committed:
[skip ci] minor

philbucher commented:
I added some more stuff to the python-scripts
@tteschemacher there are some things that should be updated, we should discuss it tmr

tteschemacher committed:
Updates in IgaIoProcess

tteschemacher committed:
Rename iga_rhino_output_process.py -> iga_output_process

Conflicts:
	applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
	applications/IgaApplication/python_scripts/iga_io_process.py
	applications/IgaApplication/python_scripts/iga_output_process.py
	applications/IgaApplication/python_scripts/iga_rhino_output_process.py
	applications/IgaApplication/tests/test_iga_rhino_output_process.py

tteschemacher committed:
Merge remote-tracking branch 'origin/feature/iga_brep_topology_kernel' into feature/iga_brep_topology_kernel

tteschemacher committed:
iga_output_process

tteschemacher committed:
iga_output_process

tteschemacher committed:
PlotIntegrationDomain

tteschemacher committed:
Updates in Plot Integration Domain. (Save of figure possible now)

tteschemacher committed:
Updates in python test functions

tteschemacher committed:
Updates in topology classes

tteschemacher committed:
New Conditions:
Point Support and Load

tteschemacher committed:
Output tests

tteschemacher committed:
Test functions for iga_output_process.py

tteschemacher committed:
SURFACE_NORMAL neglected. (->NORMAL)

Codacy changes.

tteschemacher committed:
Codacy review changes.

tteschemacher committed:
iga output process

tteschemacher committed:
Updates in topology

tteschemacher committed:
updates for Rhino integration

philbucher committed:
some renamings

philbucher committed:
merge master

philbucher committed:
bugfix in iga-solver

philbucher committed:
solving some compilation-errors

philbucher commented:
this PR is really messy now, it won't be merged, separate PRs will be made

tteschemacher committed:
CMakeList update

tteschemacher committed:
New Conditions: Coupling and Surface Load

tteschemacher committed:
bug_fixes

tteschemacher committed:
New geometry utilities for conditions and elements

tteschemacher committed:
Bug-fixes

tteschemacher committed:
New Load Curve Condition

tteschemacher committed:
New Variable types

tteschemacher committed:
small bug fixes

tteschemacher committed:
Updates

dagmawibekel committed:
Corrected a typo in the include path

Typo caused include errors in Linux since includes are case sensitive in contrast to Windows

tteschemacher committed:
New conditions

tteschemacher committed:
Updates

tteschemacher committed:
Merge branch 'feature/iga_brep_topology_kernel' of https://github.com/KratosMultiphysics/Kratos into feature/iga_brep_topology_kernel

tteschemacher committed:
Embedded Edges

tteschemacher committed:
Updates for Kratos Geometry Reading

tteschemacher committed:
Updates in Analysis Stage python files

tteschemacher committed:
Merge branch 'master' of https://github.com/KratosMultiphysics/Kratos into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
#	applications/IgaApplication/iga_application.cpp

tteschemacher committed:
small updates

tteschemacher committed:
updates in python scripts

tteschemacher committed:
remove print output

philbucher commented:
this can be closed now right?

tteschemacher committed:
New functionalities

dagmawibekel committed:
Solved compilation errors only occuring in linux

tteschemacher committed:
Reverse ShapeOptimization

tteschemacher committed:
process_factory master merge

tteschemacher committed:
Merge branch 'feature/iga_brep_topology_kernel' of https://github.com/KratosMultiphysics/Kratos.git into feature/iga_brep_topology_kernel

tteschemacher committed:
Merge branch 'feature/iga_brep_topology_kernel' of https://github.com/KratosMultiphysics/Kratos.git into feature/iga_brep_topology_kernel

tteschemacher committed:
gitignore

tteschemacher committed:
new processes
formfinding
move_mesh_process
output reaction

tteschemacher committed:
VS

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/ShapeOptimizationApplication/test_examples/01_Strain_Energy_Minimization_3D_Hook/optimization_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/01_Strain_Energy_Minimization_3D_Hook/primal_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/02_Strain_Energy_Minimization_3D_Shell/primal_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/03_mesh_independent_optimization_3D_Shell/primal_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/04_Eigenfrequency_Maximization_3D_Cantilever/optimization_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/06_bead_optimization_plate/analysis_parameters.json
#	applications/ShapeOptimizationApplication/test_examples/07_optimization_plate_with_step_adaption/analysis_parameters.json
#	applications/ShapeOptimizationApplication/tests/algorithm_bead_optimization_test/analysis_parameters.json
#	applications/ShapeOptimizationApplication/tests/algorithm_penalized_projection_test/parameters.json
#	applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/parameters.json
#	applications/ShapeOptimizationApplication/tests/algorithm_trust_region_test/parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_eigenfrequency_test/optimization_parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_shell_test/optimization_parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_solid_test/optimization_parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_step_adaption_test/parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_weighted_eigenfrequency_test/optimization_parameters.json

tteschemacher committed:
New Create function für Condition

tteschemacher committed:
Test Message

tteschemacher committed:
Updates

tteschemacher committed:
Test framework cpp tests

tteschemacher committed:
updates

tteschemacher committed:
Pressure Loads on lines

tteschemacher committed:
Merge branch 'master'

Conflicts:
	applications/IgaApplication/custom_elements/iga_truss_element.cpp
	applications/IgaApplication/custom_elements/iga_truss_element.h
	applications/IgaApplication/custom_elements/shell_kl_discrete_element.h
	applications/IgaApplication/python_scripts/assign_vector_variable_and_constraints_to_conditions_process.py
	applications/IgaApplication/python_scripts/iga_analysis.py
	applications/IgaApplication/python_scripts/iga_solver.py
	applications/IgaApplication/python_scripts/iga_static_solver.py
	applications/IgaApplication/python_scripts/python_solvers_wrapper_iga.py
	applications/ShapeOptimizationApplication/CMakeLists.txt
	applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h
	applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h
	applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_improved_integration.h
	applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h
	applications/ShapeOptimizationApplication/custom_utilities/mesh_controller_utilities.h
	applications/ShapeOptimizationApplication/python_scripts/algorithm_bead_optimization.py
	applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py
	applications/ShapeOptimizationApplication/python_scripts/mesh_controller_with_solver.py
	applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py
	applications/ShapeOptimizationApplication/shape_optimization_application.cpp
	applications/ShapeOptimizationApplication/shape_optimization_application.h
	applications/ShapeOptimizationApplication/test_examples/01_Strain_Energy_Minimization_3D_Hook/3D_Hook.mdpa
	applications/ShapeOptimizationApplication/test_examples/02_Strain_Energy_Minimization_3D_Shell/3D_Shell.mdpa
	applications/ShapeOptimizationApplication/test_examples/03_mesh_independent_optimization_3D_Shell/3D_Shell_crosswise.mdpa
	applications/ShapeOptimizationApplication/test_examples/03_mesh_independent_optimization_3D_Shell/3D_Shell_diagonal.mdpa
	applications/ShapeOptimizationApplication/test_examples/03_mesh_independent_optimization_3D_Shell/3D_Shell_longitudinal.mdpa
	applications/ShapeOptimizationApplication/test_examples/04_Eigenfrequency_Maximization_3D_Cantilever/solid_cantilever.mdpa
	applications/ShapeOptimizationApplication/test_examples/05_Constrained_Shell_Opt/3D_Shell.mdpa
	applications/ShapeOptimizationApplication/test_examples/06_bead_optimization_plate/plate.mdpa
	applications/ShapeOptimizationApplication/test_examples/07_optimization_plate_with_step_adaption/plate.mdpa
	applications/ShapeOptimizationApplication/tests/algorithm_bead_optimization_test/plate.mdpa
	applications/ShapeOptimizationApplication/tests/algorithm_penalized_projection_test/3D_Shell.mdpa
	applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/algorithm_trust_region_test/3D_Shell.mdpa
	applications/ShapeOptimizationApplication/tests/mapper_test/plate_with_quads.mdpa
	applications/ShapeOptimizationApplication/tests/mapper_test/plate_with_trias.mdpa
	applications/ShapeOptimizationApplication/tests/mapper_test/run_test.py
	applications/ShapeOptimizationApplication/tests/opt_process_eigenfrequency_test/run_test.py
	applications/ShapeOptimizationApplication/tests/opt_process_eigenfrequency_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/opt_process_shell_test/run_test.py
	applications/ShapeOptimizationApplication/tests/opt_process_shell_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/opt_process_solid_test/primal_parameters.json
	applications/ShapeOptimizationApplication/tests/opt_process_solid_test/run_test.py
	applications/ShapeOptimizationApplication/tests/opt_process_solid_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/opt_process_step_adaption_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/tent.mdpa
	applications/ShapeOptimizationApplication/tests/opt_process_weighted_eigenfrequency_test/run_test.py
	applications/ShapeOptimizationApplication/tests/opt_process_weighted_eigenfrequency_test/structure.mdpa
	applications/ShapeOptimizationApplication/tests/shape_optimization_test_factory.py
	applications/ShapeOptimizationApplication/tests/test_ShapeOptimizationApplication.py
	kratos/python_scripts/process_factory.py

tteschemacher committed:
shared -> intrusive

tteschemacher committed:
New MPM

tteschemacher committed:
updates

philbucher committed:
fixes for compilation

tteschemacher committed:
fix integration point

tteschemacher committed:
new conditions

tteschemacher committed:
Adding surface load

tteschemacher committed:
updates in load conditions

tteschemacher committed:
iga dem updates

tteschemacher committed:
bugfix in geometries

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/master' into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/ParticleMechanicsApplication/custom_python/add_custom_constitutive_laws_to_python.cpp
#	applications/ParticleMechanicsApplication/particle_mechanics_application.cpp
#	applications/ShapeOptimizationApplication/python_scripts/algorithm_penalized_projection.py
#	applications/ShapeOptimizationApplication/python_scripts/algorithm_steepest_descent.py
#	applications/ShapeOptimizationApplication/python_scripts/data_logger_factory.py
#	applications/ShapeOptimizationApplication/python_scripts/value_logger_base.py
#	applications/ShapeOptimizationApplication/python_scripts/value_logger_bead_optimization.py
#	applications/ShapeOptimizationApplication/python_scripts/value_logger_penalized_projection.py
#	applications/ShapeOptimizationApplication/python_scripts/value_logger_steepest_descent.py
#	applications/ShapeOptimizationApplication/python_scripts/value_logger_trust_region.py
#	applications/ShapeOptimizationApplication/tests/algorithm_trust_region_test/run_test.py
#	kratos/geometries/geometry.h

tteschemacher committed:
revert prints

loumalouomega committed:
Adding delaunator-cpp to external_libraries

loumalouomega committed:
Moving

loumalouomega committed:
Adding tests (WIP)

loumalouomega committed:
Adding big case (manually)

loumalouomega committed:
Adding Volodymyr Bilonenko as author

loumalouomega committed:
Removing not necessary in final version

loumalouomega committed:
Conflict on name

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/external_lib/delaunatorcpp' into feature/iga_brep_topology_kernel

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/particle/big-cleanups' into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/ParticleMechanicsApplication/custom_python/add_custom_utilities_to_python.cpp
#	applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py
#	applications/ParticleMechanicsApplication/python_scripts/mpm_static_solver.py

loumalouomega committed:
Revert "Merge branch 'master' into external_lib/delaunatorcpp"

This reverts commit e841d7e25a27cf48510750cfd48a127c0df3cd4d.

loumalouomega committed:
Merge branch 'master' into external_lib/delaunatorcpp

loumalouomega committed:
Solving issue with multiple definitions

loumalouomega committed:
Revert "Solving issue with multiple definitions"

This reverts commit 399f8df838ce1461111f1d3af4afd3fe3ecd2756.

loumalouomega committed:
Adding DelaunatorUtilities

loumalouomega committed:
Adding test for DelaunatorUtilities

loumalouomega committed:
Exposing to python

loumalouomega committed:
Fix

loumalouomega committed:
Fix conflicts

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/external_lib/delaunatorcpp' into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/ShapeOptimizationApplication/python_scripts/analyzer_factory.py
#	applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py
#	applications/ShapeOptimizationApplication/python_scripts/communicator_factory.py
#	applications/ShapeOptimizationApplication/python_scripts/optimizer_factory.py

tteschemacher committed:
Adding missing variables

philbucher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

philbucher committed:
fixing tobis shitty code

philbucher committed:
adding compiling version of CouplingInterfaceMapper

tteschemacher committed:
updates towards CAD integration in core

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/feature/iga_brep_topology_kernel' into feature/iga_brep_topology_kernel

tteschemacher committed:
Updates

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/core/nurbs_geometry' into feature/iga_brep_topology_kernel

tteschemacher committed:
updates

loumalouomega committed:
[Core][Proposal] Adding pSetGeometry to Geometrical object

Trying to used the coupled geometry I was unable to set the geometry in the conditions. Though in adding the following method

loumalouomega committed:
Using on test

loumalouomega committed:
Merge branch 'master' into core/proposal-pSetGeometry

loumalouomega committed:
Merge branch 'master' into core/proposal-pSetGeometry

loumalouomega committed:
Merge branch 'master' into core/proposal-pSetGeometry

loumalouomega committed:
Merge branch 'master' into core/proposal-pSetGeometry

tteschemacher committed:
small changes

tteschemacher committed:
Merge branch 'master' into feature/iga_brep_topology_kernel

# Conflicts:
#	applications/IgaApplication/CMakeLists.txt
#	applications/IgaApplication/custom_conditions/load_condition.cpp
#	applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp
#	applications/IgaApplication/custom_python/iga_python_application.cpp
#	applications/IgaApplication/custom_utilities/node_surface_geometry_3d.h
#	applications/IgaApplication/iga_application.cpp
#	applications/IgaApplication/iga_application_variables.cpp
#	applications/IgaApplication/iga_application_variables.h
#	applications/ShapeOptimizationApplication/ShapeOptimizationApplication.py
#	applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h
#	applications/ShapeOptimizationApplication/custom_utilities/optimization_utilities.h
#	applications/ShapeOptimizationApplication/custom_utilities/search_based_functions.h
#	applications/ShapeOptimizationApplication/python_scripts/algorithm_factory.py
#	applications/ShapeOptimizationApplication/python_scripts/analyzer_empty.py
#	applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py
#	applications/ShapeOptimizationApplication/python_scripts/custom_math.py
#	applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py
#	applications/ShapeOptimizationApplication/python_scripts/design_logger_unv.py
#	applications/ShapeOptimizationApplication/python_scripts/design_logger_vtk.py
#	applications/ShapeOptimizationApplication/python_scripts/mapper_factory.py
#	applications/ShapeOptimizationApplication/python_scripts/mesh_controller_basic_updating.py
#	applications/ShapeOptimizationApplication/python_scripts/mesh_controller_with_solver.py
#	applications/ShapeOptimizationApplication/tests/algorithm_bead_optimization_test/optimization_parameters.json
#	applications/ShapeOptimizationApplication/tests/algorithm_bead_optimization_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/algorithm_penalized_projection_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/algorithm_steepest_descent_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/opt_process_step_adaption_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/parameters.json
#	applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/opt_process_weighted_eigenfrequency_test/run_test.py
#	applications/ShapeOptimizationApplication/tests/trust_region_projector_test/run_test.py
#	kratos/geometries/geometry.h
#	kratos/geometries/geometry_data.h
#	kratos/geometries/geometry_dimension.h
#	kratos/geometries/geometry_shape_function_container.h
#	kratos/geometries/nurbs_curve_geometry.h
#	kratos/geometries/nurbs_shape_function_utilities/nurbs_curve_shape_functions.h
#	kratos/geometries/nurbs_shape_function_utilities/nurbs_surface_shape_functions.h
#	kratos/geometries/nurbs_surface_geometry.h
#	kratos/python/add_utilities_to_python.cpp
#	kratos/sources/kratos_application.cpp

tteschemacher committed:
Merge remote-tracking branch 'remotes/origin/core/proposal-pSetGeometry' into feature/iga_brep_topology_kernel

tteschemacher committed:
updates within DEM

tteschemacher committed:
updates for IGA DEM

tteschemacher committed:
revert merge
